homeassistant/components/sensor/tado_v1.py

```diff
@@ -0,0 +1,258 @@
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the sensor platform."""
+    #  pylint: disable=W0613
+
+    #  get the PyTado object from the hub component
+    tado = hass.data['Mytado']
+
+    try:
+        zones = tado.getZones()
+    except RuntimeError:
+        _LOGGER.error("Unable to get zone info from mytado")
+        return False
+
+    tado_data = TadoData(tado, MIN_TIME_BETWEEN_SCANS)
+
+    sensor_items = []
+    for zone in zones:
+        if zone['type'] == 'HEATING':
+            for variable in SENSOR_TYPES:
+                sensor_items.append(tado_data.create_zone_sensor(
+                    zone, zone['name'], zone['id'], variable))
+
+    me_data = tado.getMe()
+    sensor_items.append(tado_data.create_device_sensor(
+        me_data, me_data['homes'][0]['name'],
+        me_data['homes'][0]['id'],
+        "tado bridge status"))
+
+    tado_data.update()
+
+    if len(sensor_items) > 0:
+        add_devices(sensor_items)
+        return True
+    else:
+        return False
+
+
+class TadoSensor(Entity):
+    """Representation of a tado Sensor."""
+
+    def __init__(self, tado_data, zone_name, zone_id, zone_variable, data_id):
+        """Initialization of TadoSensor class."""
+        self._tado_data = tado_data
+        self.zone_name = zone_name
+        self.zone_id = zone_id
+        self.zone_variable = zone_variable
+        self._unique_id = '{} {}'.format(zone_variable, zone_id)
+        self._data_id = data_id
+
+        self._state = None
+        self._state_attributes = None
+
+    @property
+    def should_poll(self):
+        """Polling needed for tado sensors."""
+        return True
+
+    @property
+    def unique_id(self):
+        """Return the unique id."""
+        return self._unique_id
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self.zone_name, self.zone_variable)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        return self._state_attributes
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        if self.zone_variable == "temperature":
+            return TEMP_CELSIUS
+        elif self.zone_variable == "humidity":
+            return '%'
+        elif self.zone_variable == "heating":
+            return '%'
+
+    @property
+    def icon(self):
+        """Icon for the sensor."""
+        if self.zone_variable == "temperature":
+            return 'mdi:thermometer'
+        elif self.zone_variable == "humidity":
+            return 'mdi:water-percent'
+
+    def update(self):
+        """Update method called when should_poll is true."""
+        self._tado_data.update()
+
+        self.push_update(self._tado_data.get_data(self._data_id), True)
```

Comment:
what exactly do you mean? the `_tado_data.update()` and `_tado_data.get_date(..)`-stuff?